src/linter/ui5Types/fix/collections/sapUiCoreFixes.ts

```diff
@@ -112,9 +113,370 @@ t.declareModule("sap/ui/core/Core", [
+			moduleName: "sap/ui/core/Component",
+			validateArguments: (ctx, fixHints, vComponent) => {
+				ctx.json = "";
+				if (vComponent?.kind === SyntaxKind.ObjectLiteralExpression) {
+					let asyncOption = false;
+					ts.forEachChild(vComponent, function (node: ts.Node) {
+						if (ts.isPropertyAssignment(node) && ts.isIdentifier(node.name) && node.name.text === "async") {
+							if (node.initializer.kind === SyntaxKind.TrueKeyword) {
+								asyncOption = true; // Migration is possible, async loading is enabled
+							} else if (ts.isIdentifier(node.initializer)) {
+								const {checker} = fixHints;
+								const type = checker.getTypeAtLocation(node.initializer);
+								if ("intrinsicName" in type && type.intrinsicName === "true") {
+									asyncOption = true;
+								}
+							}
+						} else if (ts.isShorthandPropertyAssignment(node) &&
+							ts.isIdentifier(node.name) && node.name.text === "async") {
+							const {checker} = fixHints;
+							const actualSymbol = checker.getShorthandAssignmentValueSymbol(node);
+
+							if (actualSymbol?.valueDeclaration &&
+								ts.isVariableDeclaration(actualSymbol.valueDeclaration) &&
+								actualSymbol.valueDeclaration.initializer?.kind === SyntaxKind.TrueKeyword) {
+								asyncOption = true;
+							}
+						}
+					});
+
+					if (asyncOption) {
+						const parts: string[] = [];
+						const extractJson = (node: ts.Node) => {
+							ts.forEachChild(node, function (node: ts.Node) {
+								if (ts.isPropertyAssignment(node) && ts.isIdentifier(node.name)) {
+									const name = node.name.getFullText().trim();
+									if (ts.isStringLiteralLike(node.initializer) || ts.isIdentifier(node.initializer)) {
+										parts.push(`${name}: ${node.initializer.getFullText().trim()}`);
+									}
+								} else if (ts.isShorthandPropertyAssignment(node) && ts.isIdentifier(node.name)) {
+									parts.push(`${node.getFullText().trim()}`);
+								}
+							});
+						};
+						extractJson(vComponent);
+						ctx.json = `{${parts.join(", ")}}`;
+					}
+					return asyncOption;
+				} else {
+					return false; // Migration is not possible (sync component creation)
+				}
+			},
+			generator: (ctx, [moduleIdentifier]) => {
+				return `${moduleIdentifier}.create(${ctx.json})`;
+			},
+		})),
+		// Parameter bAsync has to be omitted or set to false since the new API returns
+		// the resource bundle synchronously. When bAsync is true, the new API is not a replacement
+		// as it does not return a promise. In an await expression, it would be okay, but otherwise not.
+		// sLibrary must be a library.
+		t.method("getLibraryResourceBundle", callExpressionGeneratorFix({
+			moduleName: "sap/ui/core/Lib",
+			validateArguments: (ctx: {fallback: string}, fixHints, arg1, arg2, arg3) => {
+				// Handling fallback in the legacy API
+				if (!arg1 ||
+					(ts.isIdentifier(arg1) && arg1.text === "undefined") ||
+					(ts.isStringLiteralLike(arg1) && arg1.text === "undefined")) {
```

Comment:
Checking for a string with the value `undefined` seems odd to me. Is this correct?

Reply:
Got it, but thats the line above, right? `(ts.isIdentifier(arg1) && arg1.text === "undefined")`

I was wondering about  `(ts.isStringLiteralLike(arg1) && arg1.text === "undefined")`, which I think would only match a string like `"undefined"`

Reply:
To be honest, I really don't recall this case 
I'll remove it for now